// mailinator-client/lib/message/GetInboxRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetInboxRequest = void 0;
const MailinatorRestClient_1 = require("../MailinatorRestClient");
const Constants_1 = require("../Constants");
const _resolveTemplateUrl = (domain, inbox) => {
    if (inbox === undefined) {
        inbox = '';
    }
    return `https://api.mailinator.com/v2/domains/${domain}/inboxes/${inbox}`;
};
class GetInboxRequest {
    constructor(domain, inbox, skip, limit, sort, decodeSubject) {
        this.domain = domain;
        this.inbox = inbox;
        this.skip = skip;
        this.limit = limit;
        this.sort = sort;
        this.decodeSubject = decodeSubject;
    }
    execute(apiToken) {
        const _options = {
            queryParameters: {
                params: {}
            },
            additionalHeaders: {
                [Constants_1.AUTHORIZATION]: apiToken
            }
        };
        if (this.skip !== undefined) {
            _options.queryParameters.params['skip'] = this.skip;
        }
        if (this.limit !== undefined) {
            _options.queryParameters.params['limit'] = this.limit;
        }
        if (this.sort !== undefined) {
            _options.queryParameters.params['sort'] = this.sort;
        }
        if (this.decodeSubject !== undefined) {
            _options.queryParameters.params['decode_subject'] = this.decodeSubject.toString();
        }
        return MailinatorRestClient_1.default.get(_resolveTemplateUrl(this.domain, this.inbox), _options);
    }
}
exports.GetInboxRequest = GetInboxRequest;
